### PR TITLE
feat(metrics): Add option to send hostname with every metric

### PR DESCRIPTION
**Pull request by loewenheim**
This is an adaptation of https://github.com/getsentry/relay/pull/625/files. This is intended to address https://getsentry.atlassian.net/browse/NATIVE-129.

### CODE REVIEW

**Review by Swatinem — APPROVED** [review]

**Comment by loewenheim** [comment]
Incidentally, is there a way to test this? Would it even make sense to do so?

**Comment by Swatinem** [comment]
the way to test these metrics things would be to set up a local docker image will all the relevant servers in there, which is a bit of a pain in the rear.

File: crates/symbolicator/src/metrics.rs [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 use std::net::ToSocketAddrs;
+use std::ops::{Deref, DerefMut};
 use std::sync::Arc;
 
-use cadence::{StatsdClient, UdpMetricSink};
+use cadence::{Metric, MetricBuilder, StatsdClient, UdpMetricSink};
 use parking_lot::RwLock;
 
+type Hostname = String;
+type HostnameTag = String;
```
Comment (flub) [inline crates/symbolicator/src/metrics.rs:10]: i'm curious why you chose aliases rather than newtypes?

Comment (loewenheim) [reply crates/symbolicator/src/metrics.rs:10]: I just wanted it to be clear which string is which in the definition of `MetricsClient`. I don't think a newtype would provide any extra safety in this situation.

**Pull request merged**